pkg/component/controller/konnectivityagent.go

```diff
@@ -103,11 +106,26 @@ func (k *KonnectivityAgent) writeKonnectivityAgent(clusterConfig *v1beta1.Cluste
 	if err != nil {
 		return err
 	}
+
+	proxyServerHost := k.APIServerHost
+	proxyServerPort := clusterConfig.Spec.Konnectivity.AgentPort
+
+	// We don't use k0snet.ParseHostPort here because the API server host might be an IP
+	// literal (IPv6). We don't want to change the current behavior and fail here. So we
+	// just use the standard library function and change default values only if we successfully parsed host and port.
+	host, port, _ := net.SplitHostPort(k.APIServerHost)
+	if host != "" {
+		proxyServerHost = host
+	}
+	if p, _ := strconv.Atoi(port); p != 0 {
+		proxyServerPort = int32(p)
+	}
```

Comment:
I wonder if this is the same as:


```suggestion
	serverEndpoint, err := k0snet.ParseHostPortWithDefault(
		cmp.Or(clusterConfig.Spec.Konnectivity.ExternalAddress, k.APIServerHost),
		uint16(clusterConfig.Spec.Konnectivity.AgentPort),
	)
	if err != nil {
		return fmt.Errorf("failed to determine server endpoint: %w", err)
	}
```